pyproject.toml

```diff
@@ -21,7 +21,7 @@ classifiers = [
 requires-python = ">=3.9"
 dynamic = ["version"]
 dependencies = [
-    "xarray",
+    "xarray@git+https://github.com/TomNicholas/xarray#egg=concat-no-indexes",
```

Comment:
This was required to get a few tests to pass. Even with this change I am still getting the error

```
E       NotImplementedError: ManifestArrays can't be converted into numpy arrays or pandas Index objects
```

from virtualizarr/tests/test_xarray.py:175 TestConcat.test_concat_dim_coords_along_existing_dim

Reply:
The CI in this PR is not using the correct branch of xarray - the CI show the error is raised on this line

```python
result = type(datasets[0])(result_vars, attrs=result_attrs)
```

but in https://github.com/pydata/xarray/pull/8872 and my `concat-no-indexes` branch [this line is different](https://github.com/TomNicholas/xarray/blob/cb70bec455eae709137136ba8ae9f2044e699cdd/xarray/core/concat.py#L669)

the line should read 

```python
result = type(datasets[0])(result_vars, coords=coords, attrs=result_attrs)
```

so something is up with the pip-installing of that specific branch.

(I'm trying to get https://github.com/pydata/xarray/pull/8872 merged so that we can just use xarray `main`)

Reply:
Also note that this test passes on the CI on `main` of this package

https://github.com/TomNicholas/VirtualiZarr/actions/runs/8457581269/job/23169875291

Reply:
ok trying with the default xarray branch, like in the main branch of this package.